### ISSUE TITLE
dcxwebctrl.cpp updates?

### ISSUE THREAD

**Issue by warrenarea** [twig/dcxdll#11]
I'm wondering if there was any updates to the webctrl that weren't included
in this build? 

When compiled, the "status" messages keep returning this:
D_ERROR DcxWebControl::Invoke() (D_CERROR (webctrl) xdid DcxWebControl::Invoke() uStats 100: Unable to get object state: 0 262825416)

any ideas on what I can check, or how i can fix this?

on a seperate matter, closing a dialog window while a webctrl is open, seems to cause mIRC to crash.

**Comment by OokEek** [comment]
I looked at this crash in the C++14 branch, & found it only happens when IE11 is set as the default browser for mIRC (using the registry entry as shown on a previous post), but not when using IE7.

I fixed this in that branch (sort of) by commenting out the SafeRelease() call in the web controls destructor, as it seems to be a double free issue.

**Comment by warrenarea** [comment]
yep, that fixed it for the IE11 setup.
		this->m_pWebBrowser2->Quit();
		//	this->m_pWebBrowser2->Release( );

i tried setting it back to IE7, and it seems to crash as soon as the page finishes loading.
regardless of whether i commented out release or not, IE7 seemed to crash.

I don't plan on using IE7 though so that doesn't interest me as much.

The "Unable to get object state" is a bit of an issue for me though, because I'm using Javascript 
to change window.status to send mIRC back evaluated javascript code from a page.

i suppose i could just be making calls directly back through the dcx Marked alias directly, 
and bypass the window.status altogether.

**Comment by warrenarea** [comment]
just found this too . . .

[IHTMLWindow2::execScript is no longer supported. Starting with Internet Explorer 11, use eval. For info, see Compatibility changes.]

https://msdn.microsoft.com/en-us/library/aa741364%28v=vs.85%29.aspx

line 262 dcxwebctrl.cpp

**Comment by OokEek** [comment]
I've made changes to webctrl to fix this, give the latest versions a try.

**Comment by warrenarea** [comment]
I seem to have ran into a bit of a snag with your C++14 branch.
Your master branch supported VC++ below 2013.

I am currently using VC++ 2012.
I tried to work around this by having Visual Studio upgrade (or downgrade) the project,
however, I discovered in your defines.h it is explicitly looking for VS 2015

**Comment by OokEek** [comment]
Only VS 2015+ supports C++14, VS 2015 Community is a free download from Microsoft.

**Comment by warrenarea** [comment]
So I acquired VS 2015 Community, I did run into a few minor compile issues.

```
This project references NuGet package(s) that are missing on this computer
```
The solution was to open up both the project files in a text editor
and delete the Nuget <target> header, at the bottom of the file.
This was probably because the Nuget settings wern't updated properly
during VS 2015 install. 

Because my project folder had a space in it, it wouldn't create the 
gitbuild.h in the SourceControl project. 

I solved this by changing the Nmake build command line, adding quotes 
to the "$($ProjectDir)GIT-VS-VERSION.bat"

I also had to add quotes to GIT-VS-VERSION.bat
```
SET CACHE_FILE="%~s1\GIT_VERSION"
SET HEADER_OUT_FILE="%2"
```

I am now updating assemblies to Windows 10 SDK
as I believe it now relies on that framework.
